src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManager.cs

```diff
@@ -168,7 +168,8 @@ public virtual int StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsH
         /// <summary>
         /// Cancels the test run.
         /// </summary>
-        public virtual void Cancel()
+        /// <param name="eventHandler"> EventHandler for handling execution events from Engine. </param>
+        public virtual void Cancel(ITestRunEventsHandler eventHandler)
         {
             // Cancel fast, try to stop testhost deployment/launch
```

Comment:
Based on our assumption everywhere else, that is Cancel comes before StartTestRun.
If Cancel comes, it will result in a HandleTestRunComplete, where we have a baseTestRunEventsHandler which only gets set if Start was called, else it could also potentially cause null reference.